apptest/tests/metricsql_test.go

```diff
@@ -97,9 +97,8 @@ func testInstantQueryDoesNotReturnStaleNaNs(t *testing.T, sut apptest.Prometheus
 
 	got = sut.PrometheusAPIV1Query(t, "metric[2m]", "2024-01-01T00:02:00.000Z", "5m", opts)
 	want = apptest.NewPrometheusAPIV1QueryResponse(t, `{"data": {"result": [{"metric": {"__name__": "metric"}, "values": []}]}}`)
-	s := make([]*apptest.Sample, 2)
+	s := make([]*apptest.Sample, 1)
 	s[0] = apptest.NewSample(t, "2024-01-01T00:01:00Z", 1)
-	s[1] = apptest.NewSample(t, "2024-01-01T00:02:00Z", decimal.StaleNaN)
```

Comment:
Since your PR fixes the bug, this is correct to remove this line from expectations.

Also please fix the comment for this check. Lines 94-96 should be changed to:

// Verify that instant query with default rollup function does not include stale NaN.

Reply:
Yes, I have changed the comment.